### DISCUSSION TITLE
SDK v2 release date

### DISCUSSION THREAD

**Discussion by MMarus** [Azure/azure-iot-sdk-csharp#3470] [Q&A] [answered]
Dear team,
I would like to ask when (roughly) could we expect the v2 SDK release or was the v2 release canceled?
I wasn't able to find too much information about this.
Any info would be very helpful, thank you very much and wish you a wonderful day.

**Comment by timtay-microsoft** [comment, marked as answer]
Unfortunately, we don't currently plan on releasing the v2 SDK. Nor do we plan on putting out more preview releases for the v2 SDK.

**Reply by dartasen** [reply]
What's the reason behind that ?

**Comment by pierre-sk** [comment]
Hello @timtay-microsoft,

Reopening this old discussion because I saw some recent activity around SDK v2 (#3515), that’s a good news.
Just wondering if there’s any rough idea or timeline for the next release?

Thanks a lot!

**Comment by timtay-microsoft** [comment]
Our team has finally prioritized releasing this v2 SDK after all! I have made a few changes to the v2 since the last preview release (primarily bumped dependency versions + replacing newtonsoft with System.text.json).

I have already publicly released some release candidate versions of the SDK and we are aiming to formally release it around the end of this month

https://www.nuget.org/packages/Microsoft.Azure.Devices/2.0.0-rc
https://www.nuget.org/packages/Microsoft.Azure.Devices.Client/2.0.0-rc
https://www.nuget.org/packages/Microsoft.Azure.Devices.Provisioning.Client/2.0.0-rc
https://www.nuget.org/packages/Microsoft.Azure.Devices.Provisioning.Service/2.0.0-rc

**Reply by dartasen** [reply]
That's some awesome news

**Reply by pierre-sk** [reply]
Great news, thanks!

**Reply by timtay-microsoft** [reply]
Actually, plans are changing a bit. We do still plan on releasing new versions of this SDK (and the other language SDKs!), but we are re-designing them more broadly now. We no longer plan to formally release the above RC versions, but we are planning to release something even better later.

**Reply by mmoles-Growlink** [reply]
What sort of timeline are we looking at?   3 Months, 6 months , 12 months, or  ??? months because yall are still trying to define requirements?   

I know whatever code i write now, will need to be rewritten, so getting some idea is really helpful.  I would hate to code something now, just to see 2.0 pop up next month.
